File: src/chisel/checker/models/result.py
from __future__ import annotations

from dataclasses import dataclass, field

from chisel.checker.models.violation import Violation


@dataclass(frozen=True, slots=True)
class CheckResult:
    violations: list[Violation] = field(default_factory=list)
    errors: int = 0
    warnings: int = 0
    info: int = 0
    files_checked: int = 0

    @property
    def has_errors(self) -> bool:
        return self.errors > 0

File: chisel_js/src/chisel/checker/models/result.ts
import type { Severity } from "./severity";
import type { Violation } from "./violation";

export interface CheckResult {
  readonly violations: readonly Violation[];
  readonly errors: number;
  readonly warnings: number;
  readonly info: number;
  readonly filesChecked: number;
  readonly hasErrors: boolean;
}

export function createCheckResult(
  violations: Violation[],
  filesChecked: number
): CheckResult {
  const errors = violations.filter((v) => v.severity === "error").length;
  const warnings = violations.filter((v) => v.severity === "warning").length;
  const info = violations.filter((v) => v.severity === "info").length;
  return Object.freeze({
    violations: Object.freeze([...violations]),
    errors,
    warnings,
    info,
    filesChecked,
    hasErrors: errors > 0,
  });
}
